TickedPriorityQueue/Source/Implementation/TickedQueueItem.cs

```diff
@@ -45,10 +53,12 @@ public TickedQueueItem(ITicked ticked, DateTime currentTime, bool isLooped = tru
 		{
             if (ticked == null) throw new ArgumentNullException("Missing a valid ITicked reference");
             _ticked = ticked;
+            _creationId = nextCreationId++;
```

Comment:
This change is always indexing a TickedQueueItem by _creationId. The _creationId only gets assigned on initialization, and as far as I can see, never again.   We then sort by _creationId;

However, TickedQueue _can_ re-add items on looping, in which case the just-added items should go to the end. Need to verify that this is actually happening, see if the tests are covering the case.